lib/Sema/TypeCheckAttr.cpp

```diff
@@ -7217,6 +7228,20 @@ void AttributeChecker::visitUnsafeNonEscapableResultAttr(
   }
 }
 
+void AttributeChecker::visitStaticExclusiveOnlyAttr(
+    StaticExclusiveOnlyAttr *attr) {
+  if (!Ctx.LangOpts.hasFeature(Feature::StaticExclusiveOnly)) {
+    diagnoseAndRemoveAttr(attr, diag::attr_static_exclusive_only_disabled);
+  }
+
+  // Can only be applied to structs.
+  auto structDecl = cast<StructDecl>(D);
```

Comment:
Could you please add more test-cases where the new attribute is used incorrectly i.e. on `class` and `func` etc.? I don't remember whether invalid use is just a diagnostic or we strip it way completely but it would be good to have it tested regardless.